controllers/provider-aws/pkg/controlplane/internal/helper.go

```diff
@@ -0,0 +1,235 @@
+// Copyright (c) 2019 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package internal
+
+import (
+	"fmt"
+	"path/filepath"
+
+	apisaws "github.com/gardener/gardener-extensions/controllers/provider-aws/pkg/apis/aws"
+	"github.com/gardener/gardener-extensions/controllers/provider-aws/pkg/aws"
+	awsimagevector "github.com/gardener/gardener-extensions/controllers/provider-aws/pkg/imagevector"
+	extensionscontroller "github.com/gardener/gardener-extensions/pkg/controller"
+	"github.com/gardener/gardener-extensions/pkg/controller/controlplane"
+	"github.com/gardener/gardener-extensions/pkg/util"
+
+	gardencorev1alpha1 "github.com/gardener/gardener/pkg/apis/core/v1alpha1"
+	extensionsv1alpha1 "github.com/gardener/gardener/pkg/apis/extensions/v1alpha1"
+	"github.com/gardener/gardener/pkg/operation/common"
+	"github.com/gardener/gardener/pkg/utils/chart"
+	"github.com/gardener/gardener/pkg/utils/imagevector"
+	"github.com/gardener/gardener/pkg/utils/secrets"
+	"github.com/pkg/errors"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apiserver/pkg/authentication/user"
+)
+
+// Object names
+const (
+	cloudControllerManagerDeploymentName = "cloud-controller-manager"
+	cloudControllerManagerServerName     = "cloud-controller-manager-server"
+)
+
+// Helper is a control plane actuator helper type.
+type Helper struct {
```

Comment:
Don't create a `struct` - Please create functions for each one of them without this `Helper` struct.
This means e.g. for `Secrets`, split up as follows:
`Secrets`:
* `DeploySecrets`
* `DeleteSecrets`

Reply:
The `Helper` is a way to inject stuff in the `Actuator` that can be replaced by a mock implementation for testability.  If I replace it with functions, what do I inject so that it's still testable?

Reply:
Injection isn't Java-only, it's the most meaningful way in any language, including Go, to make your code testable. At Uber, where I spent some time before coming back to SAP, injection is pervasive in Go code, they even have a special framework for this. You can check my talk at a Go meetup https://www.youtube.com/watch?v=LDGKQY8WJEM and the resources mentioned there.
Using this approach all Go code I wrote at Uber was covered by tests 80% or more. The SEN project I am currently working on uses the same approach and has a similar level of test coverage (without the framework, it's not that important but the approach itself is). 
I don't think that the suggested approach is the best one. I can do whatever you ask but we are missing the chance to do better.

Reply:
Yes, however having functions as seperated allows using them easily in other code without having to instantiate your actuator / 'pollute' one's struct with the embedded one.